lineapy/plugins/pipeline_writers.py

```diff
@@ -141,47 +142,129 @@ def docker_template_name(self) -> str:
+            ):
+                DAG_TEMPLATE = load_plugin_template(
+                    "airflow_dag_PythonOperatorPerArtifact.jinja"
+                )
+                task_functions = []
+                task_definitions = []
+                for session_artifacts in self.session_artifacts_sorted:
+                    task_functions += [
+                        nc.safename
+                        for nc in session_artifacts.artifact_nodecollections
+                    ]
+                    task_definitions += [
+                        get_task_definition(nc, self.pipeline_name)
+                        for nc in session_artifacts.artifact_nodecollections
+                    ]
+                dependencies = {
+                    task_functions[i + 1]: {task_functions[i]}
+                    for i in range(len(task_functions) - 1)
+                }
+                task_graph = TaskGraph(
+                    nodes=task_functions,
+                    mapping={f: f for f in task_functions},
+                    edges=dependencies,
+                )
+                full_code = DAG_TEMPLATE.render(
+                    DAG_NAME=self.pipeline_name,
+                    MODULE_NAME=self.pipeline_name + "_module",
+                    OWNER=self.dag_config.get("owner", "airflow"),
+                    RETRIES=self.dag_config.get("retries", 2),
+                    START_DATE=self.dag_config.get(
+                        "start_date", "days_ago(1)"
+                    ),
+                    SCHEDULE_INTERVAL=self.dag_config.get(
+                        "schedule_interval", "*/15 * * * *"
+                    ),
+                    MAX_ACTIVE_RUNS=self.dag_config.get("max_active_runs", 1),
+                    CATCHUP=self.dag_config.get("catchup", "False"),
+                    task_definitions=task_definitions,
+                    tasks=task_functions,
+                    task_dependencies=task_graph.get_airflow_dependencies(
+                        setup_task="setup", teardown_task="teardown"
+                    ),
+                )
 
             # Write out file
+            full_code = prettify(full_code)
             file = self.output_dir / f"{self.pipeline_name}_dag.py"
-            file.write_text(full_code)
-            logger.info("Generated DAG file")
+            file.write_text(prettify(full_code))
+            logger.info("Generated DAG file %s", file)
         else:
             raise ValueError(
                 f'"{dag_flavor}" is an invalid airflow dag flavor.'
             )
+
+
+def get_task_definition(
+    nc: NodeCollection, pipeline_name: str, indentation=4
+) -> str:
+    """
+    Add deserialization of input variables and serialization of output
+    variables logic of the call_block and wrap them into a new function
+    definition.
+    """
+    input_var_loading_block = [
+        f"{var} = pickle.load(open('/tmp/{pipeline_name}/variable_{var}.pickle','rb'))"
```

Comment:
if this is a public method, we'll need to validate pipeline_name. one way to do this is use aubhro's slugify method that he added to utils and ensure that the result is same as the input. if not, raise an excpetion that name is invalid or something.